client/lib/analytics/statsd.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+
+function generateUrl( featureSlug: string, eventType: string, eventValue: string ) {
+	const slug = featureSlug.replace( /[.:-]/g, '_' );
+	const type = eventType.replace( '-', '_' );
+	const json = JSON.stringify( {
+		beacons: [ `calypso.${ config( 'boom_analytics_key' ) }.${ slug }.${ type }:${ eventValue }` ],
+	} );
+
+	const [ encodedSlug, jsonData ] = [ slug, json ].map( encodeURIComponent );
+	return `https://pixel.wp.com/boom.gif?v=calypso&u=${ encodedSlug }&json=${ jsonData }`;
+}
+
+export function statsdTimingUrl( featureSlug: string, eventType: string, duration: number ) {
+	return generateUrl( featureSlug, eventType, `${ duration }|ms` );
+}
+
+export function statsdCountingUrl( featureSlug: string, eventType: string, increment: number ) {
+	return generateUrl( featureSlug, eventType, `${ increment }|c` );
```

Comment:
So the `pixel.wp.com/boom.gif` URLs for recording timings and countings are exactly the same, except the unit suffix at the end? `123|ms` vs `123|c`?

The timings are recorded to Graphite with the `stats.timers.beacon.calypso`. Does the `c` unit change that to `stats.counts.beacon.calypso` as expected? I don't know how and where the `pixel.wp.com/boom.gif` code is implemented, but @josephscott  should know everything about it 🙂

Reply:
> So the `pixel.wp.com/boom.gif` URLs for recording timings and countings are exactly the same, except the unit suffix at the end? `123|ms` vs `123|c`?

That's my understanding of the docs, but that's also partly why I wanted a review from someone familiar with `statsd` 🙂 @josephscott: does this look good to you, before I submit the stats system to my code? 😅 (Also, what would be the best way of testing that the stats are indeed being correctly collected?)

Reply:
@jsnajdr: Confirmed, it should be enough. I'll see if I can run an end-to-end test to validate somehow.

Reply:
@jsnajdr I just ran an end-to-end test, and the metrics do indeed show up in `stats.counts.beacon.calypso`, as expected.